src/helpers/gateway/mod.rs

```diff
@@ -116,6 +126,38 @@ impl<T: Transport> Gateway<T> {
                 .get_or_create(channel_id, || self.transport.receive(channel_id)),
         )
     }
+
+    fn create_idle_tracker(senders: Arc<GatewaySenders>, receivers: Arc<GatewayReceivers<T>>) -> Option<tokio::task::JoinHandle<()>> {
+        Some(tokio::task::spawn(async move {
+            // Perform some periodic work in the background
+            loop {
+                let _ = tokio::time::sleep(Duration::from_secs(5)).await;
+                if senders.check_idle_and_reset() && receivers.check_idle_and_reset() {
+                    // TODO: print something
+                    let sender_message =senders.get_all_missing_records();
```

Comment:
lets delegate the responsibility of making the string here. i.e. the API should probably look like this

```rust
   let missing_records = senders.get_pending(); // fn get_pending(&self) -> impl Iterator<item = usize>
   let str = missing_records.map(...)
```

Reply:
Sure